packages/app-mobile/components/NoteEditor/CodeMirror/RegionSpec.ts

```diff
@@ -0,0 +1,153 @@
+/**
+ * Describes an inline region's formatting. For example, an italicized region.
+ */
+
+import { Text as DocumentText, SelectionRange } from '@codemirror/state';
+
+export default class RegionSpec {
+	// Patterns to use when creating an instance of the tag.
+	// E.g.
+	//   templateStart=**
+	//   templateStop=**
+	//  would be used to create
+	//   **content here**
+	public templateStart: string;
+	public templateStop: string;
+
+	// Size of the buffer used to check for a match.
+	// Particularly useful for [startExp], [stopExp].
+	// Only used when the selection is empty.
+	protected startBuffSize: number;
+	protected stopBuffSize: number;
+
+	// Regular expressions for matching possible starting/stopping
+	// regions.
+	protected startExp?: RegExp;
+	protected stopExp?: RegExp;
```

Comment:
What does "stopExt?" mean? All class members are optional anyway, aren't they?

Reply:
`stopExp` stands for `stopExpression` (reflecting that `stopExp` is a `RegExp`). This is to distinguish `startExp` and `stopExp` from `startTemplate` and `stopTemplate`.